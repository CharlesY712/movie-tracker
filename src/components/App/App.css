.App {
  display: flex;
}

.header {
  display: block;
  text-align: center;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

.works {
  width: 100%;
}
