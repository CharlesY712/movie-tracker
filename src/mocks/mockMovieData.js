module.exports = {results: [{
adult:false,
id:337167,
overview: "Believing they have left behind shadowy figures from their past, newlyweds Christian and Ana fully embrace an inextricable connection and shared life of luxury. But just as she steps into her role as Mrs. Grey and he relaxes into an unfamiliar stability, new threats could jeopardize their happy ending before it even begins.",
popularity:572.203541,
poster_path:"/jjPJ4s3DWZZvI4vw8Xfi4Vqa1Q8.jpg",
release_date:"2018-02-07",
title:"Fifty Shades Freed",
video:false,
vote_average:6.1,
vote_count:1097
},
{
adult:false,
id:284054,
overview:"King T'Challa returns home from America to the reclusive, technologically advanced African nation of Wakanda to serve as his country's new leader. However, T'Challa soon finds that he is challenged for the throne by factions within his own country as well as without.  Using powers reserved to Wakandan kings, T'Challa assumes the Black Panther mantel to join with girlfriend Nakia, the queen-mother, his princess-kid sister,  members of the Dora Milaje (the Wakandan special forces), and an American secret agent, to prevent Wakanda from being dragged into a world war.",
popularity:329.754985,
poster_path:"/uxzzxijgPIY7slzFvMotPv8wjKA.jpg",
release_date:"2018-02-13",
title:"Black Panther",
video:false,
vote_average:7.4,
vote_count:3477
}]}
