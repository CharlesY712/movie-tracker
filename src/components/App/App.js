import React, { Component } from 'react';
import './App.css';
import SignIn from '../SignIn/SignIn';
import * as movieData from '../../helper/helper'
import { Switch, Route } from 'react-router-dom';


class App extends Component {
  constructor() {
    super()
    this.state = {
      movies:[]
    }
  }

  fetchMovies = async () => {
    const movies = await movieData.fetchRecentMovies();
    this.setState({movies})
  }
  componentDidMount() {
    this.fetchMovies();
  }

  render() {
    return (
      <div className="App">
        <header>MovieTracker</header>
        <Switch>
          <Route path="/" component={SignIn} />
        </Switch>
      </div>
    );
  }
}

export default App;
