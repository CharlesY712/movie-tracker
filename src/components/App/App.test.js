import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow } from 'enzyme';
import { cleanData } from '../../mocks/mockMovieData';

describe('App', () => {
  let wrapper, mockFetchMovies;

  beforeEach(() => {
    mockFetchMovies = jest.fn()
    wrapper = shallow(
      <App fetchMovies={mockFetchMovies}/>, {disableLifecycleMethods: true});
  });

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
})





