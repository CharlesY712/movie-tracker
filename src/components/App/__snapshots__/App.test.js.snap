// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    fetchMovies={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header>
          MovieTracker
</header>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          to="/favorites"
>
          Favorites
</NavLink>,
        <Switch>
          <Route
                    exact={true}
                    path="/"
                    render={[Function]}
          />
          <Route
                    path="/login"
                    render={[Function]}
          />
          <Route
                    path="/account"
                    render={[Function]}
          />
          <Route
                    path="/favorites"
                    render={[Function]}
          />
</Switch>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "MovieTracker",
        },
        "ref": null,
        "rendered": "MovieTracker",
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "Favorites",
          "to": "/favorites",
        },
        "ref": null,
        "rendered": "Favorites",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              exact={true}
              path="/"
              render={[Function]}
/>,
            <Route
              path="/login"
              render={[Function]}
/>,
            <Route
              path="/account"
              render={[Function]}
/>,
            <Route
              path="/favorites"
              render={[Function]}
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/login",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/account",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/favorites",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header>
            MovieTracker
</header>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            to="/favorites"
>
            Favorites
</NavLink>,
          <Switch>
            <Route
                        exact={true}
                        path="/"
                        render={[Function]}
            />
            <Route
                        path="/login"
                        render={[Function]}
            />
            <Route
                        path="/account"
                        render={[Function]}
            />
            <Route
                        path="/favorites"
                        render={[Function]}
            />
</Switch>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "MovieTracker",
          },
          "ref": null,
          "rendered": "MovieTracker",
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Favorites",
            "to": "/favorites",
          },
          "ref": null,
          "rendered": "Favorites",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                exact={true}
                path="/"
                render={[Function]}
/>,
              <Route
                path="/login"
                render={[Function]}
/>,
              <Route
                path="/account"
                render={[Function]}
/>,
              <Route
                path="/favorites"
                render={[Function]}
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "path": "/login",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "path": "/account",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "path": "/favorites",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
